test/Prototypes/string_prototype/StringComparison.swift

```diff
@@ -58,23 +58,88 @@ where
       // be affected by churn in future Unicode versions.
       //
       // TODO: cost/benefit investigation of more complex check here...
-      return codeUnit < 0xD800
+      return codeUnit < 0xD800 || codeUnit > 0xDFFF
```

Comment:
I adjusted the surrounding comments in my latest push. Sorry, I've been trying to wrap everything up today before heading out, and the history got a little chaotic at the end.